app/controllers/concerns/user_roles.rb

```diff
@@ -0,0 +1,24 @@
+module UserRoles
```

Comment:
To make this easier to understand where data comes from, it might be better to turn this into a class with static / class methods on it and pass in the required data for each class method

Reply:
Hm. So something like a RoleChecker that would take a user's roles and have the foo_project_ids methods on it? 

ProjectPolicy still needs those, are you suggesting that the `editor?`/`approver?`/`viewer?` logic move to the new class as well? That'd change how the rest of the policies work, too, since they're sending project ids to a ProjectPolicy.

Reply:
I don't think so, rather the policy object uses the `RoleChecker` class methods and passes in the state that method requires.

This would remove this include module and instead you would call the `RoleChecker` class methods in place of these methods in your policy `editor?` etc methods.

Reply:
RoleChecker, checking in to check roles